### PR TITLE
Reject invalid review audit PR lists

### PR DESCRIPTION
**Pull request by Phazzie**
## Summary
- reject malformed `--prs` entries instead of silently dropping them
- prevent all-empty `--prs` input from falling back to the default PR sweep
- add regression coverage for valid `#154,166`, malformed, non-positive, and empty PR lists

## Review-thread context
Addresses the still-valid PR #154 review thread about invalid PR list entries being silently ignored:
- `PRRT_kwDOPyrwIM6LHK5e`

## Validation
- `npm run test:review-thread-audit-options`
- `npm run --silent review:unresolved -- --repo Phazzie/FairytaleswithSpice --prs 154,abc --json` exits with `Invalid --prs value: abc`
- `node --check scripts/recovery/list-unresolved-review-threads.mjs`
- `npm run --silent review:unresolved -- --repo Phazzie/FairytaleswithSpice --prs 154 --outdated-only --json`
- `git diff --check`
- `npm run test:all`
- `scripts/recovery/check-vercel-function-count.sh` (`11/12`)

`npm run recovery:preflight -- review-audit-invalid-prs --dry-run` was attempted, but the slice name is not registered in `scripts/recovery/slice-preflight.sh`; this is a tooling-slice-only change.

## Summary by Sourcery

Enforce strict validation of the --prs option for review audit scripts and add regression tests for invalid PR lists.

Bug Fixes:
- Reject malformed, non-numeric, non-positive, and empty --prs lists instead of silently ignoring invalid entries.

Tests:
- Add regression tests covering valid, malformed, non-positive, and empty --prs argument combinations for the review audit CLI.

### CODE REVIEW

File: scripts/recovery/list-unresolved-review-threads.mjs [hunk 1]
```diff
@@ -124,12 +124,29 @@ function validateOptions(options) {
...
+      if (number <= 0) {
+        throw new Error(`Invalid --prs value: ${entry}`);
+      }
+
+      return number;
+    });
+
+  if (pullRequestNumbers.length === 0) {
+    throw new Error('--prs requires at least one pull request number');
+  }
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply scripts/recovery/list-unresolved-review-threads.mjs:147]: Fixed in the current PR #174 head 0e858cf. `parsePrList()` no longer filters empty entries before validation; it preserves the friendly all-empty error for values like `,,`, but mixed or trailing empty entries such as `154,,166` and `154,` now throw `Invalid --prs value: empty entry`. Focused tests and direct CLI checks pass. Resolving as fixed.

File: tests/review-thread-audit-options.test.ts [hunk 2]
```diff
@@ -96,6 +97,19 @@ async function main() {
...
+    /Invalid --prs value: abc/,
+  );
+  assert.throws(
+    () => parseArgs(['--prs', '154,0']),
+    /Invalid --prs value: 0/,
+  );
+  assert.throws(
+    () => parseArgs(['--prs', ',,']),
+    /--prs requires at least one pull request number/,
+  );
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply tests/review-thread-audit-options.test.ts:112]: Fixed in the current PR #174 head 0e858cf. `tests/review-thread-audit-options.test.ts` now covers both `154,,166` and `154,` and asserts the invalid empty-entry path. Resolving as fixed.

File: scripts/recovery/list-unresolved-review-threads.mjs [hunk 3]
```diff
@@ -124,12 +124,29 @@ function validateOptions(options) {
...
 function parsePrList(value) {
-  return value
+  const pullRequestNumbers = value
     .split(',')
-    .map((entry) => entry.trim().replace(/^#/, ''))
+    .map((entry) => entry.trim())
     .filter(Boolean)
-    .map(Number)
-    .filter((entry) => Number.isInteger(entry) && entry > 0);
+    .map((entry) => {
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply scripts/recovery/list-unresolved-review-threads.mjs]: Fixed in the current PR #174 head 0e858cf. Empty entries are no longer silently dropped; mixed/trailing empty entries now throw `Invalid --prs value: empty entry`, while all-empty input keeps the friendly `--prs requires at least one pull request number` message. Resolving as fixed.

File: tests/review-thread-audit-options.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Phazzie) [reply tests/review-thread-audit-options.test.ts:120]: Fixed in the current PR #174 head 0e858cf. Added regression coverage for `154,,166` and `154,` in the existing parseArgs/parsePrList assertions. Resolving as fixed.

File: scripts/recovery/list-unresolved-review-threads.mjs [hunk 4]
```diff
@@ -124,12 +124,29 @@ function validateOptions(options) {
 }
 
 function parsePrList(value) {
-  return value
+  const pullRequestNumbers = value
     .split(',')
-    .map((entry) => entry.trim().replace(/^#/, ''))
+    .map((entry) => entry.trim())
     .filter(Boolean)
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply scripts/recovery/list-unresolved-review-threads.mjs]: Fixed in the current PR #174 head 0e858cf. The parser now validates each comma-separated segment and rejects empty entries instead of filtering them out before validation. Resolving as fixed.
